en/docs/deploy/api-microgateway/service-discovery.md

```diff
@@ -1,140 +1,91 @@
 # Service Discovery
 
-When you create an API Microgateway, the API Microgateway is capable of routing traffic to the URL of the API that you provide at the time of creating an API. The micro services have dynamic URLs. As the API Microgateway is immutable, the dynamic URL changes will not get reflected on the Microgateway. Therefore, in such a scenario you need to stop the Microgateway and restart it by providing the new URL. However, as the latter mentioned scenario is not practical in a real-world scenario, the API Microgateway is integrated with [etcd](https://github.com/etcd-io/etcd) for the purpose of discovering the target endpoints in order to route API traffic.
+WSO2 Microgateway supports service discovery by connecting to [Consul service registry](https://www.hashicorp.com/products/consul) and discover upstream services automatically.
```

Comment:
@rumeshmadhusanka , shall we have the following as the introduction in order to provide the user with more details.

In a microservices environment, the set of running service endpoints are not static. A service usually has multiple endpoints. Therefore, service discovery services such as Consul, which is a service mesh solution developed by HashiCorp, are used to identify which endpoints run which services. 

WSO2 API Microgateway can be used as an ingress gateway in an environment that uses Consul as a service registry, so that the APIs or services can be exposed to developers or API consumers while providing security, rate limiting, and other QoS.

Therefore, WSO2 Microgateway supports service discovery for the Envoy based Microgateway using the [Consul service registry](https://www.hashicorp.com/products/consul) so that upstream services can be discovered automatically.

Reply:
Great. 
Further, do we still use the phrase "Envoy based Microgateway" ? [see](https://github.com/wso2/docs-apim/pull/3155#discussion_r574275876)